test/support/page_objects/common.js

```diff
@@ -83,8 +85,21 @@ export default class Common {
     function navigateTo(url) {
       return self.try(function () {
         // since we're using hash URLs, always reload first to force re-render
-        self.debug('navigate to: ' + url);
-        return self.remote.get(url)
+        return esClient.getDefaultIndex()
+        .then(function (defaultIndex) {
+          if (appName === 'discover' || appName === 'visualize' || appName === 'dashboard') {
+            if (!defaultIndex) {
+              // https://github.com/elastic/kibana/issues/7496
+              // Even though most tests are using esClient to set the default index, sometimes Kibana clobbers
+              // that change.  If we got here, fix it.
+              self.debug(' >>>>>>>> WARNING Navigating to [' + appName + '] with defaultIndex=' + defaultIndex);
+              self.debug(' >>>>>>>> Setting defaultIndex to "logstash-*""');
+              esClient.updateConfigDoc({'dateFormat:tz':'UTC', 'defaultIndex':'logstash-*'});
```

Comment:
Does `updateConfigDoc` return a promise?  If so, I think we should return that promise and then do the `self.remote.get()` call in a followup `.then()` block.